src/Storages/StorageSet.cpp

```diff
@@ -229,8 +242,24 @@ void registerStorageSet(StorageFactory & factory)
                 "Engine " + args.engine_name + " doesn't support any arguments (" + toString(args.engine_args.size()) + " given)",
                 ErrorCodes::NUMBER_OF_ARGUMENTS_DOESNT_MATCH);
 
-        return StorageSet::create(args.relative_data_path, args.table_id, args.columns, args.constraints, args.context);
-    });
+        const auto & settings = args.context.getSettingsRef();
```

Comment:
Why support both query-level and table-level settings?
Usage of query-level settings is inconvenient for CREATE TABLE.

Reply:
I'd stick with table-level settings.

Reply:
I have pushed a new commit that only use table-level settings. And also add functional tests.